plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -2353,10 +2353,10 @@ protected function persist_save( &$order, bool $force_all_fields = false, $backf
 			$order->set_date_modified( current_time( 'mysql' ) );
 		}
 
-		$this->update_order_meta( $order );
-
 		$this->persist_order_to_db( $order, $force_all_fields );
 
+		$this->update_order_meta( $order );
```

Comment:
No, changing the order so that `update_order_meta` comes after `persist_order_to_db` is necessary so that the new refund object has an ID to attach the meta to. Otherwise the meta rows are generated but they have an `order_id` value of `0`.